### DISCUSSION TITLE
What is the correct sequencing of QLayout, QHeader, QToolbar, ...?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#11444] [General - Components / Directives / etc] [answered]
I am wrinting a simple app, with a toolbar and some generated content. I thought that the proper sequence for the various elements would be:

```
<q-layout>
  <q-header>
    <q-toolbar>
      <q-btn icon="plus"></q-btn>
      <q-input
        (...)
      </q-input>
      <q-icon :color="mobileStatusColor" class="q-ml-xs" name="check_circle" size="lg"></q-icon>
    </q-toolbar>
  </q-header>
  <q-page-container>
  <!-- enumerated dynamic elements -->
  </q-page-container>
<q-layout>
```

What I get is

![image](https://user-images.githubusercontent.com/1345886/142850262-f9afb2bd-8cdf-4ab0-861d-c2f24de15124.png)

Specifically, it looks like the page container goes below the toolbar:

![image](https://user-images.githubusercontent.com/1345886/142851536-9404b2c7-c2f8-42a7-ba6e-6d24f62c063f.png)

**Before going further: is my sequencing of the different layers of QLayout, QHeader, ... correct?**

**Comment by pdanpdan** [comment]
I think you are missing the `QPage` that should be inside `QPageContainer`

**Reply by wsw70** [reply]
Thanks, I added it but no changes

```
 <q-page-container>
  <q-page>
    <!-- enumerated dynamic elements -->
   </q-page>
  </q-page-container>
 
```

**Comment by pdanpdan** [comment]
Please make a codepen

**Comment by wsw70** [comment, marked as answer]
I managed to "solve" the problem. It seems that it was related to the fact that the QHeader had a `v-if` condition (and a `v-else` for another QHeader). I moved the condition one level down (to QToolbar) and the layout is correct.

The only difference is that QHeader is fixed and depending on conditions, one of two Toolbars is selected. Before it was the case for QHeader.

This is weird but solved the issue (in case someone would have the same problem someday)
